packages/grafana-toolkit/src/cli/tasks/plugin.ci.ts

```diff
@@ -0,0 +1,77 @@
+import { Task, TaskRunner } from './task';
+import { pluginBuildRunner } from './plugin.build';
+import { useSpinner } from '../utils/useSpinner';
+import { restoreCwd } from '../utils/cwd';
+
+// @ts-ignore
+import execa = require('execa');
+import path = require('path');
+import fs = require('fs');
+
+export interface PluginCIOptions {
+  dryRun?: boolean;
+}
+
+const calcJavascriptSize = (base: string, files?: string[]): number => {
+  files = files || fs.readdirSync(base);
+  let size = 0;
+
+  if (files) {
+    files.forEach(file => {
+      const newbase = path.join(base, file);
+      const stat = fs.statSync(newbase);
+      if (stat.isDirectory()) {
+        size += calcJavascriptSize(newbase, fs.readdirSync(newbase));
+      } else {
+        if (file.endsWith('.js')) {
+          size += stat.size;
+        }
+      }
+    });
+  }
+  return size;
+};
+
+const pluginCIRunner: TaskRunner<PluginCIOptions> = async ({ dryRun }) => {
+  const start = Date.now();
+  const distDir = `${process.cwd()}/dist`;
+  const artifactsDir = `${process.cwd()}/artifacts`;
+  await execa('rimraf', [`${process.cwd()}/coverage`]);
+  await execa('rimraf', [artifactsDir]);
+
+  // Do regular build process
+  await pluginBuildRunner({ coverage: true });
+  const elapsed = Date.now() - start;
+
+  if (!fs.existsSync(artifactsDir)) {
+    fs.mkdirSync(artifactsDir);
+  }
+
+  // TODO? can this typed from @grafana/ui?
+  const pluginInfo = require(`${distDir}/plugin.json`);
+  const zipName = pluginInfo.id + '-' + pluginInfo.info.version + '.zip';
+  const zipFile = path.resolve(artifactsDir, zipName);
+  process.chdir(distDir);
+  await execa('zip', ['-r', zipFile, '.']);
+  restoreCwd();
+
+  const stats = {
+    startTime: start,
+    buildTime: elapsed,
+    jsSize: calcJavascriptSize(distDir),
```

Comment:
I think this could be picked up from webpack stats. Maybe the pluginBuildRunner could expose those.

Reply:
good idea -- I'll look into it 👍

Reply:
looked into them -- we could scan some of them, but I think there is value in looking at the actual files on disk.  In particular if/when the final bundle is a group of files built in different steps.  Yes this `jsSize` call is a dummy :)